src/representation_theory/basic.lean

```diff
@@ -83,12 +83,25 @@ lemma as_algebra_hom_of (g : G):
   (as_algebra_hom ρ (of k G g)) = ρ g :=
 by simp only [monoid_algebra.of_apply, as_algebra_hom_single]
 
+/--
+If `ρ : representation k G V`, then `ρ.as_module` is a type synonym for `V`,
+which we equip with an instance `module (monoid_algebra k G) ρ.as_module`.
+You should use `as_module_equiv : ρ.as_module ≃+ V` to translate terms.
+-/
+@[nolint unused_arguments, derive add_comm_monoid]
+def as_module (ρ : representation k G V) := V
+
+instance : inhabited ρ.as_module := ⟨0⟩
+
 /--
 A `k`-linear representation of `G` on `V` can be thought of as
 a module over `monoid_algebra k G`.
 -/
-noncomputable def as_module : module (monoid_algebra k G) V :=
-  module.comp_hom V (as_algebra_hom ρ).to_ring_hom
+noncomputable instance as_module_module : module (monoid_algebra k G) ρ.as_module :=
+begin
+  change module (monoid_algebra k G) V,
+  exact module.comp_hom V (as_algebra_hom ρ).to_ring_hom,
+end
```

Comment:
```suggestion
noncomputable instance as_module_module : module (monoid_algebra k G) ρ.as_module :=
module.comp_hom V (as_algebra_hom ρ).to_ring_hom
```
You can get this working again if you add more stuff to the `derive` on line 91. You need
```
@[nolint unused_arguments, derive [add_comm_monoid, module (module.End k V)]]
```